### PR TITLE
Added test for FieldCollectionTrait

### PR DESCRIPTION
**Pull request by AIeksandr**
- Installed field_collection module
- Created test fields for collection field
- Added test for FieldCollectionTrait

### CODE REVIEW

File: tests/behat/fixtures/d7/composer.json [hunk 1]
```diff
@@ -24,6 +25,7 @@
         "drupal/wysiwyg": "^2.5",
         "library/ckeditor": "^4.9"
     },
+    "minimum-stability": "dev",
```
Comment (alexdesignworks) [inline tests/behat/fixtures/d7/composer.json]: please add ` "prefer-stable": true` below if you are adding this one.

Comment (alexdesignworks) [reply tests/behat/fixtures/d7/composer.json]: but why do we need this at all? field collection module is a stable module

File: tests/behat/fixtures/d7/docroot/sites/all/modules/custom/mysite_core/mysite_core.info [hunk 2]
```diff
@@ -69,12 +70,17 @@ features[field_base][] = field_paragraph_link
 features[field_base][] = field_paragraph_media
 features[field_base][] = field_paragraph_title
 features[field_base][] = field_tags
+features[field_base][] = field_text_first
```
Comment (alexdesignworks) [inline tests/behat/fixtures/d7/docroot/sites/all/modules/custom/mysite_core/mysite_core.info]: please rename these fields to `field_fc_text_first` and `field_fc_text_second`. a simple search and replace should work

File: src/D7/FieldCollectionTrait.php [hunk 3]
```diff
@@ -69,10 +71,13 @@ public static function fieldCollectionParseFields(BeforeNodeCreateScope $scope)
 
       $fc_field_values = explode(self::fieldCollectionGetInstanceDelimiter(), $field_value);
 
+      if (count($fc_field_values) > count($fc_field_names)) {
+        throw new \Exception(sprintf('A large number of values for collection fields. Given "%s", but should be "%s" items for "%s"', count($fc_field_values), count($fc_field_names), $field_name));
```
Comment (alexdesignworks) [inline src/D7/FieldCollectionTrait.php]: `Provided more field collection values for field "%s" then expected: provided %s, but expected %s`

File: tests/behat/fixtures/d7/docroot/sites/all/modules/custom/mysite_core/mysite_core.info (hunk 2 above)
Comment (alexdesignworks) [reply tests/behat/fixtures/d7/docroot/sites/all/modules/custom/mysite_core/mysite_core.info]: ok, this actually requires more fields:

Please add `field_collection2` field to article
please add field `field_fc_text_third` and `field_fc_text_fourth` to `field_collection2` node field.

this is to the assert that the step actually allows to add values to multiple field collection fields at once:
```
	    And article content:
      | title           | field_collection:field_text_first;field_text_second | field_collection2:field_text_third;field_text_fourth |
      | Test collection | test_text1;test_text2                               |test_text3;test_text4                                |
```

File: tests/behat/fixtures/d7/composer.json (hunk 1 above)
Comment (AIeksandr) [reply tests/behat/fixtures/d7/composer.json]: It is beta version only.

**Pull request merged**